charon/src/reorder_decls.rs

```diff
@@ -101,27 +101,22 @@ impl DeclarationGroup {
     }
 
     fn make_trait_decl_group(
-        ctx: &TransCtx,
+        _ctx: &TransCtx,
         _is_rec: bool,
         gr: impl Iterator<Item = TraitDeclId::Id>,
     ) -> Self {
         let gr: Vec<_> = gr.collect();
-        let ctx = ctx.into_fmt();
         // Trait declarations often refer to `Self`, like below,
         // which means they are often considered as recursive by our
         // analysis. TODO: do something more precise. What is important
         // is that we never use the "whole" self clause as argument,
         // but rather projections over the self clause (like `<Self as Foo>::u`,
         // in the declaration for `Foo`).
```

Comment:
If you understand what this TODO refers to, I'd like us to have a test for a case like that

Reply:
This TODO is exactly the error we got here: https://github.com/AeneasVerif/charon/issues/159